contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+ * and 'kafka.' from the provided configuration. Initializes a producer 
+ * and consumer based on the given value. Queue name is driven from the 
+ * workflow. 
+ * @author preeth
+ *
+ */
+public class KafkaObservableQueue implements ObservableQueue {
+
+  private static final Logger logger = LoggerFactory.getLogger(KafkaObservableQueue.class);
+
+  private static final String QUEUE_TYPE = "kafka";
+
+  private static final String KAFKA_PRODUCER_PREFIX = "kafka.producer.";
+
+  private static final String KAFKA_CONSUMER_PREFIX = "kafka.consumer.";
+
+  private static final String KAFKA_PREFIX = "kafka.";
+
+  private String queueName;
+
+  private int pollTimeInMS;
+
+  private int longPollTimeout;
+
+  private int pollCount;
+
+  private KafkaProducer<String, String> producer;
+
+  private KafkaConsumer<String, String> consumer;
+
+  @Inject
+  public KafkaObservableQueue(String queueName, Configuration config) {
+
+    this.queueName = queueName;
+    this.pollTimeInMS = config.getIntProperty("workflow.dyno.queues.pollingInterval", 100);
+    this.pollCount = config.getIntProperty("workflow.dyno.queues.pollCount", 10);
+    this.longPollTimeout = config.getIntProperty("workflow.dyno.queues.longPollTimeout", 1000);
+    // Init Kafka producer and consumer properties
+    Properties producerProps = new Properties();
+    Properties consumerProps = new Properties();
+    consumerProps.put("group.id", queueName + "_group");
+    int randomId = new Random().nextInt(100);
+    consumerProps.put("client.id", queueName + "_consumer_"+randomId);
+    producerProps.put("client.id", queueName + "_producer_"+randomId);
+    Map<String, Object> configMap = config.getAll();
+    for (Entry<String, Object> entry : configMap.entrySet()) {
+      String key = entry.getKey();
+      String value = (String) entry.getValue();
+      if (key.startsWith(KAFKA_PREFIX)) {
+        if (key.startsWith(KAFKA_PRODUCER_PREFIX)) {
+          producerProps.put(key.replaceAll(KAFKA_PRODUCER_PREFIX, ""), value);
+        } else if (key.startsWith(KAFKA_CONSUMER_PREFIX)) {
+          consumerProps.put(key.replaceAll(KAFKA_CONSUMER_PREFIX, ""), value);
+        } else {
+          producerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+          consumerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+        }
+      }
+    }
+
+    // Init Kafka producer and consumer
+    producer = new KafkaProducer<>(producerProps);
+    consumer = new KafkaConsumer<>(consumerProps);
+    consumer.subscribe(Collections.singletonList(queueName));
+    logger.info("KafkaaObservableQueue initialized for {}", queueName);
+  }
+
+  @Override
+  public Observable<Message> observe() {
+    OnSubscribe<Message> subscriber = getOnSubscribe();
+    return Observable.create(subscriber);
+  }
+
+  @Override
+  public List<String> ack(List<Message> messages) {
+    return messages.stream().map(Message::getId).collect(Collectors.toList());
```

Comment:
This is basically a no-op, ack is used to acknowledge that the message was processed successfully, if ack is not invoked by `EventProcessor` then the message should be retried (e.g. with SQS the message will become visible after timeout), @preeth-gopalakrishnan @pctreddy let's figure out how to support this with Kafka.

Reply:
Auto-commit is set to false and currently acknowledging the message by committing programmatically.